internal/controller/datadogagent_controller.go

```diff
@@ -98,6 +98,15 @@ type DatadogAgentReconciler struct {
 // Use CiliumNetworkPolicy
 // +kubebuilder:rbac:groups=cilium.io,resources=ciliumnetworkpolicies,verbs=get;list;watch;create;update;patch;delete
 
+// Configure Appsec Gateway Integration
```

Comment:
does the operator need these permissions if the envoyFilters and Gateways are being added by the cluster agent?